internal/controller/runner_controller.go

```diff
@@ -0,0 +1,354 @@
+		log.Error(err, "unable to update Runner status")
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *RunnerReconciler) SetupWithManager(mgr ctrl.Manager, eventChan chan event.GenericEvent) error {
+	c, err := ctrl.NewControllerManagedBy(mgr).
+		For(&garmoperatorv1alpha1.Runner{}).
+		Build(r)
+	if err != nil {
+		return err
+	}
+
+	return c.Watch(&source.Channel{Source: eventChan}, &handler.EnqueueRequestForObject{})
+}
+
+func (r *RunnerReconciler) PollRunnerInstances(ctx context.Context, eventChan chan event.GenericEvent) {
+	log := log.FromContext(ctx)
+	ticker := time.NewTicker(5 * time.Second)
+	for {
+		select {
+		case <-ctx.Done():
+			log.Info("Closing event channel for runners...")
+			close(eventChan)
+			return
+		case _ = <-ticker.C:
+			log.Info("Polling Runners...")
+			err := r.EnqueueRunnerInstances(ctx, eventChan)
+			if err != nil {
+				log.Error(err, "Failed polling runner instances")
+			}
+		}
+	}
+}
+
+func (r *RunnerReconciler) EnqueueRunnerInstances(ctx context.Context, eventChan chan event.GenericEvent) error {
+	instanceClient, err := r.instanceClient()
+	if err != nil {
+		return err
+	}
+
+	pools, err := r.fetchPools(ctx)
+	if err != nil {
+		return err
+	}
+
+	if len(pools.Items) < 1 {
+		return nil
+	}
+
+	// fetching runners by pools to ensure only runners belonging to pools in same namespace are being shown
+	garmRunnerInstances, err := r.fetchRunnerInstancesByNamespacedPools(instanceClient, pools)
+	if err != nil {
+		return err
+	}
+
+	// compares garm db with RunnerCRs and deletes RunnerCRs not present in garm db
+	err = r.cleanUpNotMatchingRunnerCRs(ctx, garmRunnerInstances)
+	if err != nil {
+		return err
+	}
+
+	// triggers controller to reconcile based on instances in garm db
+	enqeueRunnerEvents(garmRunnerInstances, eventChan)
+	return nil
+}
+
+func enqeueRunnerEvents(garmRunnerInstances params.Instances, eventChan chan event.GenericEvent) {
+	for _, runner := range garmRunnerInstances {
+		runnerObj := garmoperatorv1alpha1.Runner{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      strings.ToLower(runner.Name),
+				Namespace: config.Config.Operator.WatchNamespace,
+			},
+		}
+
+		e := event.GenericEvent{
+			Object: &runnerObj,
+		}
+
+		eventChan <- e
+	}
+}
+
+func (r *RunnerReconciler) cleanUpNotMatchingRunnerCRs(ctx context.Context, garmRunnerInstances params.Instances) error {
+	runnerCRList := &garmoperatorv1alpha1.RunnerList{}
+	err := r.List(ctx, runnerCRList)
+	if err != nil {
+		return err
+	}
+
+	runnerCRNameList := slices.Map(runnerCRList.Items, func(runner garmoperatorv1alpha1.Runner) string {
+		return runner.Name
+	})
+
+	runnerInstanceNameList := slices.Map(garmRunnerInstances, func(runner params.Instance) string {
+		return strings.ToLower(runner.Name)
+	})
+
+	runnersToDelete := getRunnerDiff(runnerCRNameList, runnerInstanceNameList)
+	fmt.Println(runnersToDelete)
```

Comment:
this should be a log message - (i would have this as a debug-message `log.V(1).Info`)